src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -0,0 +1,802 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using INTPTR_INTCAST = System.Int32;
+using INTPTR_INTPTRCAST = System.IntPtr;
+using Microsoft.Win32.SafeHandles;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+using System.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Globalization;
+using System.Security;
+
+namespace System.ServiceProcess
+{
+    /// This class represents an NT service. It allows you to connect to a running or stopped service
+    /// and manipulate it or get information about it.
+    public class ServiceController : IDisposable
+    {
+        private readonly string _machineName;
+        private readonly ManualResetEvent _waitForStatusSignal = new ManualResetEvent(false);
```

Comment:
It looks like _waitForStatusSignal is really just being used as a substitute for Thread.Sleep(250) down below.  Why not just use Thread.Sleep(250) and get rid of _waitForStatusSignal entirely?  Or have I missed some other usage somewhere?

Reply:
Filed <https://github.com/dotnet/runtime/issues/44699>.

Reply:
Created dotnet/runtime#45049 to track the proposal of using NotifyServiceStatusChange to avoid polling.